garnett/trajectory.py

```diff
@@ -494,66 +502,102 @@ def types(self, value):
         self.frame_data.types = value
 
     @property
-    def positions(self):
+    def position(self):
         "Nx3 array of coordinates for N particles in 3 dimensions."
         self.load()
-        return self._raise_attributeerror('positions')
+        return self._raise_attributeerror('position')
 
-    @positions.setter
-    def positions(self, value):
+    @property
+    def positions(self):
+        """
+        Nx3 array of coordinated for N particles in 3 dimensions.
+        Deprecated alias for position.
+        """
+        self._deprecation_warning('positions', 'position')
+        try:
+            return self.position(self)
+        except TypeError:
+            return self.position
+
+    @position.setter
+    def position(self, value):
         # Various sanity checks
         try:
             value = np.asarray(value, dtype=self._dtype)
         except ValueError:
-            raise ValueError("Positions can only be set to numeric arrays.")
+            raise ValueError("position can only be set to numeric arrays.")
```

Comment:
Some of these checks should be lifted to a generic function. It doesn't necessarily have to include the attribute name in the error, so it could just be a function of the data.

Reply:
Is this something to be fixed in this PR or a seperate one?

Reply:
@klywang @gaofy95 You can define a function like the one below 

```python
def _validate_input_array(self, value, size):
    try:
        value = np.asarray(value, dtype=self._dtype)
    except ValueError:
        raise ValueError("This property can only be set to numeric arrays.")
    if not np.all(np.isfinite(value)):
        raise ValueError("Property being set must all be finite numbers.")
    elif not len(value.shape) == 2 or value.shape[1] != size:
        raise ValueError("Input array must be of shape (N,{}) where N is the number of particles.".format(size))
    return value
```
and replace that logic on every setter by calling it:
```python
@orientation.setter
def orientation(self, value):
    self._validate_input_array(value, 4)
    self.load()
    self.frame_data.orientation = value
```